File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/translate/src/main/java/com/example/translate/TranslateText.java

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/


package com.example.translate;

import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.translate.TranslateClient;
import software.amazon.awssdk.services.translate.model.TranslateTextRequest;
import software.amazon.awssdk.services.translate.model.TranslateTextResponse;
import software.amazon.awssdk.services.translate.model.TranslateException;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class TranslateText {

    public static void main(String[] args) {

        Region region = Region.US_WEST_2;
        TranslateClient translateClient = TranslateClient.builder()
                .credentialsProvider(EnvironmentVariableCredentialsProvider.create())
                .region(region)
                .build();

        textTranslate( translateClient);
        translateClient.close();
    }

    public static void textTranslate(TranslateClient translateClient) {

        try {
            TranslateTextRequest textRequest = TranslateTextRequest.builder()
                .sourceLanguageCode("en")
                .targetLanguageCode("fr")
                .text("Its a sunny day today")
                .build();

            TranslateTextResponse textResponse = translateClient.translateText(textRequest);
            System.out.println(textResponse.translatedText());

        } catch (TranslateException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/translate/src/main/kotlin/com/kotlin/translate/TranslateText.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.translate

import aws.sdk.kotlin.services.translate.TranslateClient
import aws.sdk.kotlin.services.translate.model.TranslateTextRequest

suspend fun main(){
    textTranslate()
}

suspend fun textTranslate() {

        val textRequest = TranslateTextRequest {
            sourceLanguageCode = "en"
            targetLanguageCode = "fr"
            text = "Its a sunny day today"
        }

        TranslateClient { region = "us-west-2" }.use { translateClient ->
            val textResponse = translateClient.translateText(textRequest)
            println(textResponse.translatedText)
        }
 }
